Restore Options changes (revert #1722) (#1723)
Qiskit/documentation: docs/guides/configure-error-suppression.mdx

---
title: Configure error suppression

> description: How to configure error suppression in Qiskit Runtime. (Former title: Runtime compilation)

Reviewer: ```suggestion
description: How to configure error suppression in Qiskit Runtime. (Former title - Runtime compilation)
```

> description: How to configure error suppression in Qiskit Runtime. (Previous topic - runtime compilation)

Reviewer 2: Is the parenthetical to make sure the keywords "runtime compilation" still show up in search? "Previous topic" sounds a bit like it could refer to the topic before this one. Maybe (Previously known as) or (Also known as) or something?

Reviewer: I think so - Eric made that change.  Maybe "previously titled"?

Reviewer: Former title

> description: How to configure error suppression in Qiskit Runtime. (Previous topic - runtime compilation)

Reviewer: ```suggestion
description: How to configure error suppression in Qiskit Runtime. (Former title: Runtime 
 compilation)
```

---

# Configure error suppression

Error suppression refers to techniques where you use knowledge about the undesirable effects to introduce customization that can anticipate and avoid the potential impacts of those effects. These techniques often consist of altering or adding control signals to ensure that the quantum processor returns the desired results.  This typically results in quantum pre-processing overhead; therefore, it is important to achieve a balance between perfecting your results and ensuring that your job completes in a reasonable amount of time.

> V2 primitives support a number of error suppression techniques, including dynamical decoupling and Pauli twirling. See [Error mitigation and suppression techniques](error-mitigation-and-suppression-techniques) for an explanation of each. In V2 primitives, you can turn on or off individual method. See the [Advanced error suppression options](#transpilation-table) section for details.

Reviewer 3: The error mitigation page has links to the options for ZNE etc, but DD/twirling are not linked here.

> V2 primitives support a number of error suppression techniques, including [dynamical decoupling](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.DynamicalDecouplingOptions#dynamicaldecouplingoptions) and Pauli [twirling](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.TwirlingOptions). See [Error mitigation and suppression techniques](error-mitigation-and-suppression-techniques) for an explanation of each. In V2 primitives, you can turn on or off individual method. See the [Advanced error suppression options](#transpilation-table) section for details.
> 
> Estimator V2 employs error suppression and mitigation by default. If you don't want any processing done to your input circuits follow the instructions in the [Turn off all error mitigation and error suppression](specify-runtime-options#no-error-mitigation) section.

Reviewer 2: ```suggestion
V2 primitives support a number of error suppression techniques, including [dynamical decoupling](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.DynamicalDecouplingOptions#dynamicaldecouplingoptions) and [Pauli twirling](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.TwirlingOptions). See [Error mitigation and suppression techniques](error-mitigation-and-suppression-techniques) for an explanation of each. In V2 primitives, you can turn on or off individual methods. See the [Advanced error suppression options](#transpilation-table) section for details.

Estimator V2 employs error suppression and mitigation by default. If you don't want any processing done to your input circuits, follow the instructions in the [Turn off all error mitigation and error suppression](specify-runtime-options#no-error-mitigation) section.
```

---
title: Configure error suppression
description: How to configure error suppression in Qiskit Runtime. (Previous topic - runtime compilation)

---

# Configure error suppression for Qiskit Runtime

> Runtime compilation techniques transform your circuit to minimize errors. Runtime compilation adds some classical pre-processing overhead to your overall runtime. Therefore, it is important to achieve a balance between perfecting your results and ensuring that your job completes in a reasonable amount of time.

Reviewer 3: Since we changed the topic to "error suppression", we should talk about error suppression instead of compilation (compilation is just the tool we use to apply error suppression instructions). 

Stealing words from this [article](https://www.ibm.com/quantum/blog/quantum-error-suppression-mitigation-correction), we can have an intro similar to that of error mitigation page:

"Error suppression refers to techniques where you use knowledge about the undesirable effects to introduce customization that can anticipate and avoid the potential impacts of those effects. These techniques often consist of altering or adding control signals to ensure that the quantum processor returns the desired results.

V2 primitives support a number of error suppression techniques, including dynamical decoupling and Pauli twirling. See [Error mitigation and suppression techniques](https://docs.quantum.ibm.com/guides/error-mitigation-and-suppression-techniques) for explanation of each. In V2 primitives, you can turn on or off individual method. See the [Advanced error suppression options] section for details."

> Primitives let you employ runtime compilation by choosing advanced runtime compilation options. If you don't want any processing done to minimize errors, follow the instructions in the [Turn off all error mitigation and error suppression](runtime-options-overview#no-error-mitigation) section.

Reviewer 3: Similar to above, the wording should be about error suppression and not compilation.

```suggestion
Estimator V2 employs error suppression and mitigation by default. If you don't want any processing done to your input circuits follow the instructions in the [Turn off all error mitigation and error suppression](runtime-options-overview#no-error-mitigation) section.
```

<span id="transpilation-table"></span>
## Advanced error suppression options

<Tabs>
  <TabItem value="PrimV2" label="V2 primitives">

>     In the V2 primitives, you can explicitly enable and disable individual error mitigation and suppression methods, such as dynamical decoupling.

Reviewer 3: We should link to the API reference (or the new options table) for people to find other options, like we do in the error mitigation page.

<span id="transpilation-table"></span>
## Advanced error suppression options

<Tabs>
  <TabItem value="PrimV2" label="V2 primitives">

>     In the V2 primitives, you can explicitly enable and disable individual error mitigation and suppression methods, such as dynamical decoupling. See [Advanced Qiskit Runtime options](runtime-options-overview) for information about available options.

Reviewer 3: I like this points to the overview table instead of the API reference. Maybe the error mitigation page can also do the same? (under "Custom error settings (V2 primitives)")

```suggestion
    In the V2 primitives, you can explicitly enable and disable individual error mitigation and suppression methods, such as dynamical decoupling. See [Available options](runtime-options-overview#available-options) for information about available options.
```

    <Admonition type = "note">

>     Dynamical decoupling is not supported when the input circuits are dynamic.

Reviewer 3: Similar to the error mit page, it's better to point the compatibility table instead of singling out just DD.

  </TabItem>

  <TabItem value="PrimV1" label="V1 primitives">
    Because all input must be ISA, `options.transpilation.xxx` is ignored. See [Advanced runtime options](runtime-options-overview) for information.
  </TabItem>
</Tabs>

## Turn off all error suppression

For instructions to turn off all error suppression, see the [Turn off all error suppression and mitigation](runtime-options-overview#no-error-mitigation) section.

## Next steps

<Admonition type="tip" title="Recommendations">

>     - Try a tutorial that uses optimization levels, such as the [Variational quantum eigensolver](https://learning.quantum.ibm.com/tutorial/variational-quantum-eigensolver) tutorial.

Reviewer 3: This tutorial uses `optimization_level` in the transpiler, which is unrelated to the (now deprecated) optimization_level in the primitives 😅

    - Learn how to transpile locally in the [Transpile](./transpile/) section.

>     - Try the [Submit pre-transpiled circuits](https://learning.quantum.ibm.com/tutorial/submitting-user-transpiled-circuits-using-primitives) tutorial.

Reviewer 3: Are the links under "next steps" supposed to be next steps in a user's workflow or related pages? If the former, then they are not good next steps, since transpilation happens before execution.
